pkg/select/blockdevice/filters.go

```diff
@@ -242,3 +250,59 @@ func filterNodeName(originalBD *apis.BlockDeviceList, spec *apis.DeviceClaimSpec
 	}
 	return filteredBDList
 }
+
+// filterBlockDeviceTag is used to filter out BlockDevices which do not have the
+// block-device-tag label. This filter works on a block device list which has
+// already been filtered by the given selector.
+func filterBlockDeviceTag(originalBD *apis.BlockDeviceList, spec *apis.DeviceClaimSpec) *apis.BlockDeviceList {
+
+	// if the block-device-tag label was already included in the selector
+	// given in the BDC by the user, then this filter is not required. This
+	// is because it would have already performed the filter operation with the
+	// label. If the label is not present, a new selector is made to remove
+	// devices which have that label.
+	if !isBlockDeviceTagSelectorRequired(spec.Selector) {
+		return originalBD
+	}
+
+	// a DoesNotExist requirement is created to filter out devices which have
+	// the block-device-tag label
+	req, err := labels.NewRequirement(kubernetes.BlockDeviceTagLabel, selection.DoesNotExist, []string{})
+
+	// this error should never occur, because error for DoesNotExist happen
+	// only when non zero length of values are passed
+	if err != nil {
+		klog.Info("could not create requirement for label ", kubernetes.BlockDeviceTagLabel)
+		return originalBD
+	}
+	sel := labels.NewSelector()
```

Comment:
n_: `sel` can be renamed as `deviceTagLabelDoesNotExist`.  We can make the code more readable by making use of the fact that go doesn't have limitations on the variable names.